app/jobs/monthly_interest_accrual_job.rb

```diff
@@ -1,5 +1,5 @@
-class MonthlyInterestAccrualJob < ApplicationJob
-  def perform
+class MonthlyInterestAccrualJob < TaskJob
+  def perform(_job_params)
```

Comment:
So having to add `_job_params` as an argument to perform for every TaskJob seems to be the main price to pay for the approach to arbitrary job params that's in 9573 (where we have ```def enqueue(job_params: {})
    job = job_class.constantize.perform_later(job_params.merge(task_id: id))
    self.update_attribute(:provider_job_id, job.provider_job_id)
  end```)

Reply:
args to `enqueue` are optional, but perform has to accept at least one argument if it's not used, because the Task model has to pass the task_id to perform_later. It's used in the ActiveJob callbacks implemented in TaskJob.